src/mod/plugins/zoraxy_plugin/events/events.go

```diff
@@ -82,12 +98,31 @@ func (e *AccessRuleCreatedEvent) GetName() EventName {
 	return EventAccessRuleCreated
 }
 
+func (e *AccessRuleCreatedEvent) GetEventSource() string {
+	return "accesslist-api"
+}
+
+type CustomEvent struct {
+	SourcePlugin string         `json:"source_plugin"`
+	Recipients   []string       `json:"recipients"`
+	Payload      map[string]any `json:"payload"`
+}
```

Comment:
Idea is that CustomEvents have a source, destination, and custom payload.

At the zoraxy level, we don't need to worry about the details of what that payload looks like since presumably if plugins are sending messages to eachother they can figure out their own message format.
An alternative to the `map[string]any` would be a byte array but that's a little finicky.